From 207a0519ac73290ba65b6e5f7446549a2a66f5d2 Mon Sep 17 00:00:00 2001
From: Stefan Gehn
Date: Mon, 05 Dec 2011 16:12:28 +0000
Subject: Do not free nullpointer in Darwin disk reading

Calling CFRelease() on a NULL value is wrong and causes crashes.
This should fix MacPorts ticket #29565.
---
diff --git a/src/sysdeps/darwin.c b/src/sysdeps/darwin.c
index c36788c..f506a29 100644
--- src/sysdeps/darwin.c
+++ src/sysdeps/darwin.c
@@ -450,7 +450,6 @@ gkrellm_sys_disk_read_data(void)
             gkrellm_debug(DEBUG_SYSDEP,
                 "No statistics dict found in storage driver, skipping disk '%s'\n",
                 disk->path);
-            CFRelease(storage_driver_stats);
             IOObjectRelease(storage_driver);
             continue;
         }
--
cgit v0.9.0.2-2-gbebe
